### PR TITLE
Fix regression with camera sensors not using the background color set in `<scene>`

### PR DESCRIPTION
**Pull request by azeey**
# 🦟 Bug fix

## Summary
Currently, the background of camera sensors is not affected by the value of `<background>` in `<scene>`. This appears to be a regression since Citadel.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: test/integration/camera_sensor_background_from_scene.cc [hunk 1]
```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
```
Comment (iche033) [inline test/integration/camera_sensor_background_from_scene.cc:2]: 2022

**Review by iche033 — APPROVED** [review]
works for me.

File: test/integration/camera_sensor_background_from_scene.cc (hunk 1 above)
Comment (azeey) [reply test/integration/camera_sensor_background_from_scene.cc]: Done in https://github.com/gazebosim/gz-sim/pull/1515/commits/92bab296b7fd4c2e39732e6414d45bea2fa9d2ab

**Comment by azeey** [comment]
`SensorsFixture.SensorsBatteryState` test has been failing on the ign-gazebo6 stable CI. I think it might have started in https://github.com/gazebosim/gz-sim/pull/1480

**Pull request merged**